logger.go

```diff
@@ -109,18 +120,26 @@ func New(fac zapcore.Facility, options ...Option) Logger {
 	return log
 }
 
+func (log *logger) Named(s string) Logger {
+	if s == "" {
+		return log
+	}
+	l := log.clone()
+	if log.name == "" {
+		l.name = s
+	} else {
+		l.name = strings.Join([]string{l.name, s}, ".")
```

Comment:
Instead of having the logger always use `.` as a separator, what about if we pass name around as a `[]string`, and let the `NameFormatter` decide how to convert that `[]string` into a `zapcore.Field`?